pkg/server/singleprocess/service_pipeline.go

```diff
@@ -213,6 +352,36 @@ func (s *Service) RunPipeline(
 					},
 				},
 			}
+		case *pb.Pipeline_Step_Pipeline_:
+			embeddedPipeline, err := s.state(ctx).PipelineGet(o.Pipeline.Ref)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+
+			// Pass through *this* steps job dependency IDs so that the child
+			// steps aren't scheduled prior to any dependencies they are waiting for.
```

Comment:
```suggestion
			// Pass through *this* step's job dependency IDs, so that the child
			// step's jobs aren't scheduled prior to any dependencies.
```
Is this what you mean?

Reply:
Yep!